### DISCUSSION TITLE
JSON file load error with MobileNet

### DISCUSSION THREAD

**Discussion by MOAZ47** [apache/incubator-mxnet#19481] [Q&A] [answered]
I am trying to load a pre-trained MobileNet, this is the code, 

`sym, arg_params, aux_params = mx.model.load_checkpoint('/content/mobilenet_v2', 0)`

And this is the error I see,

`/usr/local/lib/python2.7/dist-packages/mxnet/model.pyc in load_checkpoint(prefix, epoch)
    448     - Parameters will be loaded from ``prefix-epoch.params``.
    449     """
--> 450     symbol = sym.load('%s-symbol.json' % prefix)
    451     save_dict = nd.load('%s-%04d.params' % (prefix, epoch))
    452     arg_params = {}

/usr/local/lib/python2.7/dist-packages/mxnet/symbol/symbol.pyc in load(fname)
   2726         raise TypeError('fname need to be string')
   2727     handle = SymbolHandle()
-> 2728     check_call(_LIB.MXSymbolCreateFromFile(c_str(fname), ctypes.byref(handle)))
   2729     return Symbol(handle)
   2730 

/usr/local/lib/python2.7/dist-packages/mxnet/base.pyc in check_call(ret)
    251     """
    252     if ret != 0:
--> 253         raise MXNetError(py_str(_LIB.MXGetLastError()))
    254 
    255 

MXNetError: [16:49:36] /home/travis/build/dmlc/mxnet-distro/mxnet-build/3rdparty/dmlc-core/include/dmlc/././json.h:731: Check failed: ch == '{' (60 vs. {) : Error at Line 5, around ^`!DOCTYPE html>`, Expect '{' but get '<'
Stack trace:
  [bt] (0) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x2795cb) [0x7fefb1f395cb]
  [bt] (1) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x3f9ce63) [0x7fefb5c5ce63]
  [bt] (2) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x3fae7f6) [0x7fefb5c6e7f6]
  [bt] (3) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x3fa7708) [0x7fefb5c67708]
  [bt] (4) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x3fa82ba) [0x7fefb5c682ba]
  [bt] (5) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x25aceb1) [0x7fefb426ceb1]
  [bt] (6) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x3f7259d) [0x7fefb5c3259d]
  [bt] (7) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x25ba7ff) [0x7fefb427a7ff]
  [bt] (8) /usr/local/lib/python2.7/dist-packages/mxnet/libmxnet.so(+0x25aceb1) [0x7fefb426ceb1]`

Need help to fix this

**Comment by leezu** [comment]
@MOAZ47 it appears your file is invalid. It's a html file but we expect a json file.

**Reply by MOAZ47** [reply]
This is how the data looks like,

`{
  "nodes": [
    {
      "op": "null", 
      "name": "data", 
      "inputs": []
    }, 
    {
      "op": "null", 
      "name": "conv1_weight", 
      "attr": {
        "kernel": "(3, 3)", 
        "no_bias": "True", 
        "num_filter": "32", 
        "pad": "(1, 1)", 
        "stride": "(2, 2)"
      }, 
      "inputs": []
    }, 
    {
      "op": "Convolution", 
      "name": "conv1", 
      "attr": {
        "kernel": "(3, 3)", 
        "no_bias": "True", 
        "num_filter": "32", 
        "pad": "(1, 1)", 
        "stride": "(2, 2)"
      }, `

I am pretty sure this is JSON

**Reply by leezu** [reply]
Any guess why the error message says `Error at Line 5, around ^!DOCTYPE html>` then?

**Comment by MOAZ47** [comment, marked as answer]
I was trying a few things randomly, and have resolved the issue. In the beginning, I was using mxnet to download the "JSON" and "Param" file. Below is the code I used

`mx.test_utils.download(path+'blob/master/mobilenet_v2-0000.params')`

Instead of using mxnet, I did  "**! git clone https://github.com/KeyKy/mobilenet-mxnet**" to download the repo, and it worked. I don't know how but it works.
So if anyone out there can explain the difference between using mxnet to download and git clone, I will really appreciate it.

**Reply by leezu** [reply]
`blob/master/mobilenet_v2-0000.params` leads to an html file. You'll need to click the button on github "Raw" to get the raw file
